inference-engine/src/transformations/src/transformations/common_optimizations/fq_mul_fusion.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/fq_mul_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+// This transformation multiplies the "output_low" and "output_high" inputs of the FQ operation
+// by the constant value that before transormation is used to multiply the output of FQ.
+// Two new Multiply nodes share the original constant C6.
+//
+//             C1 C2 C3 C4 C5                   C1 C2 C3      C4                        C5
+//             |  |  |  |  |                    |  |  |       |                         |
+//             |  |  |  |  |                    |  |  |       v                         v
+//             v  v  v  v  v                    |  |  |  +----------+              +----------+
+//           +--------------+                   |  |  |  | Multiply | <--- C6 ---> | Multiply |
+//           | FakeQuantize |                   |  |  |  +----+-----+              +----+-----+
+//           +--------------+                   |  |  |       |                         |
+//                  |                =====>     |  |  |       |                         |
+//                  v                           v  v  v       v                         v
+//             +----------+                    +----------------------------------------------+
+//             | Multiply | <--- C6            |                 FakeQuantize                 |
+//             +----+-----+                    +----------------------------------------------+
+//                  |                                                   |
+//                  v                                                   v
+//
+
+ngraph::pass::FakeQuantizeMulFusion::FakeQuantizeMulFusion() {
+  const auto fq_output_low_p = ngraph::pattern::wrap_type<opset4::Constant>();
+  const auto fq_output_high_p = ngraph::pattern::wrap_type<opset4::Constant>();
+
+  const auto fq_node_p = ngraph::pattern::wrap_type<opset4::FakeQuantize>(
+      {ngraph::pattern::wrap_type<opset4::Constant>(),
```

Comment:
We should perform it for non-constant input too.

Reply:
You can use `auto input = pattern::any_input();` inputs instead of Constants